### PR TITLE
feat: updated tab visual design

### PR DESCRIPTION
**Pull request by SiTaggart**
Description and useful links to discussions / issues / tickets

### Change One

Details of change...

### Change Two

Details of change...

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

File: packages/paste-core/components/tabs/src/Tab.tsx [hunk 1]
```diff
@@ -56,39 +56,59 @@ const getTabBoxStyles = (orientation: Orientation, variant: Variants): BoxStyleP
...
+        /* Selected State */
         _selected: {
-          borderBottomColor: isInverse ? 'colorBorderInverseStrong' : 'colorBorderPrimary',
-          color: isInverse ? 'colorTextInverse' : 'colorTextLink',
-        },
-        _hover: {
-          borderBottomColor: isInverse ? 'colorBorderInverse' : 'colorBorderPrimaryStronger',
-          color: isInverse ? 'colorTextInverseWeak' : 'colorTextLinkStronger',
+          borderColor: isInverse ? 'colorBorderInverseStrongest' : 'colorBorderWeak',
+          borderBottomColor: isInverse ? 'colorBorderInverseWeakest' : 'colorBorderInverseStrongest',
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/Tab.tsx:77]: If we add `color-border-weakest` to our tokens, we can swap `colorBorderInverseStrongest` here for `colorBorderWeakest` and not get the white border in dark mode.

suggestion: gray-0 in twilio/default, #FFFFFF in evergreen, gray-110 in twilio-dark/dark

File: packages/paste-core/components/tabs/src/TabList.tsx [hunk 2]
```diff
@@ -17,11 +17,27 @@ export interface TabListProps {
...
+  element,
+}) => {
+  const isInverse = variant === 'inverse' || variant === 'inverse_fitted';
+
+  return (
+    <Box
+      display="flex"
+      borderBottomStyle="solid"
+      borderBottomWidth="borderWidth10"
+      borderBottomColor={isInverse ? 'colorBorderInverseWeaker' : 'colorBorderWeaker'}
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/TabList.tsx:32]:
```suggestion
      borderBottomColor={isInverse ? 'colorBorderInverseWeaker' : 'colorBorderWeak'}
```

**Comment by serifluous** [comment]
I can't figure out where line height is defined, so I'm leaving this as a general comment here:

Can you change the line height to line-height-30? It's ok if it changes vertical tabs, too.

File: packages/paste-core/components/tabs/src/Tab.tsx [hunk 3]
```diff
@@ -56,39 +56,60 @@ const getTabBoxStyles = (orientation: Orientation, variant: Variants): BoxStyleP
...
         },
+        /* Hover states */
+        _hover: {
+          color: isInverse ? 'colorTextInverse' : 'colorTextPrimary',
+          borderBottomColor: isInverse ? 'colorBorderInverseStrongest' : 'colorBorderPrimary',
+        },
+        _selected_hover: {
+          borderColor: isInverse ? 'colorBorderInverseStrongest' : 'colorBorderWeak',
+          color: isInverse ? 'colorTextInverse' : 'colorTextPrimaryStronger',
+          borderBottomColor: isInverse ? 'colorBorderInverseWeakest' : 'colorBorderInverseStrongest',
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/Tab.tsx:93]:
```suggestion
          borderBottomColor: isInverse ? 'colorBorderInverseWeakest' : 'colorBorderWeakest',
```

File: packages/paste-core/components/tabs/src/Tab.tsx [hunk 4]
```diff
@@ -56,39 +56,60 @@ const getTabBoxStyles = (orientation: Orientation, variant: Variants): BoxStyleP
...
+        bottom: '-1px',
+        /* Selected State */
         _selected: {
-          borderBottomColor: isInverse ? 'colorBorderInverseStrong' : 'colorBorderPrimary',
-          color: isInverse ? 'colorTextInverse' : 'colorTextLink',
-        },
-        _hover: {
-          borderBottomColor: isInverse ? 'colorBorderInverse' : 'colorBorderPrimaryStronger',
-          color: isInverse ? 'colorTextInverseWeak' : 'colorTextLinkStronger',
+          borderColor: isInverse ? 'colorBorderInverseStrongest' : 'colorBorderWeak',
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/Tab.tsx:76]:
```suggestion
          borderColor: isInverse ? 'colorBorderInverse' : 'colorBorderWeak',
```

File: packages/paste-core/components/tabs/src/Tab.tsx [hunk 5]
```diff
@@ -56,39 +56,60 @@ const getTabBoxStyles = (orientation: Orientation, variant: Variants): BoxStyleP
...
         },
+        /* Disabled state */
         _disabled: {
           borderBottomColor: 'transparent',
           color: isInverse ? 'colorTextInverseWeaker' : 'colorTextWeaker',
         },
+        /* Hover states */
+        _hover: {
+          color: isInverse ? 'colorTextInverse' : 'colorTextPrimary',
+          borderBottomColor: isInverse ? 'colorBorderInverseStrongest' : 'colorBorderPrimary',
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/Tab.tsx:88]: on hover:
- in default, the top/right/left border color stays the same (color-border-weak)
- in inverse, the top/right/left border color changes to color-border-inverse-strongest

File: packages/paste-core/components/tabs/src/Tab.tsx [hunk 6]
```diff
@@ -56,39 +56,60 @@ const getTabBoxStyles = (orientation: Orientation, variant: Variants): BoxStyleP
...
+          borderColor: isInverse ? 'colorBorderInverse' : 'colorBorderPrimary',
+          boxShadow: 'shadowFocusInset',
+          borderBottomColor: 'colorBorderPrimary',
+        },
+        _selected_focus: {
+          borderStyle: 'solid',
+          borderWidth: 'borderWidth10',
+          boxShadow: 'shadowFocusInset',
+          color: isInverse ? 'colorTextInverse' : 'colorTextPrimary',
+          borderColor: isInverse ? 'colorBorderInverse' : 'colorBorderPrimary',
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/Tab.tsx:107]:
```suggestion
          borderColor: 'colorBorderPrimary',
```

File: packages/paste-core/components/tabs/src/Tab.tsx [hunk 7]
```diff
@@ -56,39 +56,60 @@ const getTabBoxStyles = (orientation: Orientation, variant: Variants): BoxStyleP
...
+          borderColor: isInverse ? 'colorBorderInverseStrongest' : 'colorBorderWeak',
+          color: isInverse ? 'colorTextInverse' : 'colorTextPrimaryStronger',
+          borderBottomColor: isInverse ? 'colorBorderInverseWeakest' : 'colorBorderInverseStrongest',
+        },
+        /* Focus states */
         _focus: {
-          boxShadow: isInverse ? 'shadowFocusInverse' : 'shadowFocus',
           outline: 'none',
-          borderRadius: 'borderRadius20',
+          borderColor: isInverse ? 'colorBorderInverse' : 'colorBorderPrimary',
```
Comment (serifluous) [inline packages/paste-core/components/tabs/src/Tab.tsx:98]: not sure, but this one may need to change to just colorBorderPrimary for both inverse and non-inverse, too.

File: packages/paste-design-tokens/tokens/themes/twilio-dark/global/border-color.yml [hunk 8]
```diff
@@ -18,6 +18,11 @@ props:
     value: "{!palette-blue-90}"
     comment: Weakest primary border color
 
+  # inverse borders
+  color-border-inverse-weakest:
+    value: "{!palette-gray-100}"
+    comment: Weakest border on inverse backgrounds. Must be used on color-background-body-inverse.
```
Comment (serifluous) [inline packages/paste-design-tokens/tokens/themes/twilio-dark/global/border-color.yml:24]:
```suggestion
    comment: Weakest border on inverse backgrounds. Must be used on color-background-inverse.
```

**Review by serifluous — DISMISSED** [review]

**Review by serifluous — APPROVED** [review]

**Review by TheSisb — APPROVED** [review]

**Pull request merged**